File: 895/go/main.go
package main

import "fmt"

func Max(i, j int) int {
	if i > j {
		return i
	}
	return j
}

func main() {
	n := 0
	ans := 1
	fmt.Scanf("%d", &n)
	nums := make([]int, n)
	dp := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&nums[i])
		dp[i] = 1
	}
	for i := 1; i < n; i++ {
		for j := 0; j < i; j++ {
			if nums[j] < nums[i] {
				dp[i] = Max(dp[j]+1, dp[i])
				ans = Max(dp[i], ans)
			}
		}
	}
	fmt.Println(ans)
}

File: 895/cpp/main.cpp
#include<iostream>
#include<algorithm>
using namespace std;

const int N = 1001;
int n;
int dp[N];
int a[N];

int main () {
    int ans = 1;
    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
        dp[i] = 1;
    }
    for(int i = 1; i < n; i++) {
        for ( int j = 0; j < i; j++ ) {
            if (a[j] < a[i]) {
                dp[i] = max(dp[j]+1, dp[i]);
                ans = max(ans, dp[i]);
            }
        }
    }
    cout << ans << endl;
    return 0;
}